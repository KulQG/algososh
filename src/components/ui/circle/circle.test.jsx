import React from "react";
import renderer from "react-test-renderer";

import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe('Circle', () => {
    it('the circle is rendered without errors (without text)', () => {
        const tree = renderer
            .create(<Circle />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered without errors (with text)', () => {
        const tree = renderer
            .create(<Circle text='c4' />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered without errors (head, tail) ', () => {
        const tree = renderer
            .create(<Circle head={'text'} tail={'text'} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered without errors (head, tail: components)', () => {
        const tree = renderer
            .create(<Circle head={<Circle isSmall />} tail={<Circle isSmall />} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered without errors (index)', () => {
        const tree = renderer
            .create(<Circle index={0} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered with correct state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Changing} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered with correct state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Default} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('the circle is rendered with correct state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Modified} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})