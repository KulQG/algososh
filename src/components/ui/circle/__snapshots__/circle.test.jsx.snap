// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle the circle is rendered with correct state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle the circle is rendered with correct state 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle the circle is rendered with correct state 3`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle the circle is rendered without errors (head, tail)  1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    text
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    text
  </div>
</div>
`;

exports[`Circle the circle is rendered without errors (head, tail: components) 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle the circle is rendered without errors (index) 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle the circle is rendered without errors (with text) 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle the circle is rendered without errors (without text) 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
